docs/platform_developers/database.rst

```diff
@@ -46,5 +49,65 @@ updates:
 3. When schema updates are available, each ``update`` function between the current version and the latest version is executed
    in the right order.
 
-When a schema update fails, the database schema is rolled-back to the latest schema version for which the ``update`` function
-did succeed. In that case the Inmanta server will fail to start.
+When a schema update fails, the database schema is rolled-back to the state before the start of the Inmanta server. In
+that case the Inmanta server will fail to start.
+
+
+Testing database migrations
+###########################
+
+Each database migration script should be tested using an automated test case. The tests that verify the migration from
+schema version ``<old_version>`` to ``<new_version>`` are stored in a file named
+``tests/db/test_v<old_version>_to_v<new_version>.py``.
+
+In general, a database schema migration test has the following flow:
+
+1. Load a database dump that uses the database schema version directly preceding the version being tested.
+2. Perform assertions that verify the database schema before the migration.
+3. Start the inmanta server to trigger the database migration scripts.
+4. Perform assertions to verify that the migration was done correctly.
+
+The example below shows a test for the above-mentioned database migration script.
+
+.. code-block:: python
+    :linenos:
+
+    # File: tests/db/test_v202101010_to_v202102220.py
+    @pytest.mark.db_restore_dump(os.path.join(os.path.dirname(__file__), "dumps/v202101010.sql"))
```

Comment:
```suggestion
    @pytest.mark.db_restore_dump(os.path.join(os.path.dirname(__file__), "dumps", "v202101010.sql"))
```
I'm aware this is an issue in many of the test cases as well, but might as well do it right in the docs.